### ISSUE TITLE
preselectedOptionIndex on checkbox causes unintended selection when navigating in apps

### ISSUE THREAD

**Issue by haakemon** [Altinn/app-frontend-react#141]

**Comment by olemartinorg** [comment]
Multiple apps in production now uses this functionality, especially `fufinn-behovsendring`. I also noticed a need for improving this functionality when debugging Altinn/altinn-studio#8255. Maybe triggering this functionality should be moved out from the component, if possible. The way it is implemented now means the component will re-render itself after the first render where it has options (but none has been selected). In the mentioned issue, multiple levels of this leads to unnecessary requests (I haven't fixed that part, as it's not a breaking bug). When the added complexity of dependencies taken into account, solving this in the frontend in a way that avoid re-renders and useless requests quickly becomes unbearable, and not really worth the effort to do properly for what it brings to the table, IMO.

**Comment by olemartinorg** [comment]
As so many other issues, this related to:
- #377 
- #345
- #555

With #555 an app-developer could implement support for this behavior in repeating groups via `ProcessDataWrite` (by automatically selecting an option for new rows), and for data outside repeating groups, it is already possible with a prefill. In order to fix this problem while maintaining the functionality, we would need some way to indicate (outside the data model!) that the user made a conscious choice to revert our pre-selection and go back to a neutral state. In addition to that, the feature relies on an `index` inside the options, not a `value`, making it very fickle in cases where you don't know the order of options returned.

TL;DR: Let's implement #555, deprecate and remove the feature instead. Bat-signalling #339!
